src/DateTimeParser.jl

```diff
@@ -0,0 +1,512 @@
+                res["year"] = ymd[1]
+            else
+                res["day"] = ymd[1]
+            end
+        end
+    elseif len_ymd == 2
+        # Two members with numbers
+        if ymd[1] > 31
+            # 99-01
+            res["year"], res["month"] = ymd
+        elseif ymd[2] > 31
+            # 01-99
+            res["month"], res["year"] = ymd
+        elseif dayfirst && ymd[2] <= 12
+            # 13-01
+            res["day"], res["month"] = ymd
+        else
+            # 01-13
+            res["month"], res["day"] = ymd
+        end
+    elseif len_ymd == 3
+        # Three members
+        if mstridx == 1
+            res["month"], res["day"], res["year"] = ymd
+        elseif mstridx == 2
+            if ymd[1] > 31 || (yearfirst && ymd[3] <= 31)
+                # 99-Jan-01
+                res["year"], res["month"], res["day"] = ymd
+            else
+                # 01-Jan-01
+                # Give precendence to day-first, since
+                # two-digit years is usually hand-written.
+                res["day"], res["month"], res["year"] = ymd
+            end
+        elseif mstridx == 3
+            # WTF
+            if ymd[2] > 31
+                # 01-99-Jan
+                res["day"], res["year"], res["month"] = ymd
+            else
+                res["year"], res["day"], res["month"] = ymd
+            end
+        else
+            if ymd[1] > 31 || (yearfirst && ymd[2] <= 12 && ymd[3] <= 31)
+                # 99-01-01
+                res["year"], res["month"], res["day"] = ymd
+            elseif ymd[1] > 12 || (dayfirst && ymd[2] <= 12)
+                # 13-01-01
+                res["day"], res["month"], res["year"] = ymd
+            else
+                # 01-13-01
+                res["month"], res["day"], res["year"] = ymd
+            end
+        end
+    end
+
+    # Fill in default values if none exits
+    if !haskey(res, "year") && !haskey(res, "month") && !haskey(res, "day")
+        res["year"], res["month"], res["day"] = Dates.yearmonthday(default)
+    else
+        if haskey(res, "year")
+            res["year"] = convertyear(res["year"])
+        else
+            res["year"] = year(default)
+        end
+        if !haskey(res, "month")
+            res["month"] = 1
+        end
+        if !haskey(res, "day")
+            res["day"] = 1
+        end
+    end
+    if !haskey(res, "hour") && !haskey(res, "minute") && !haskey(res, "second") && !haskey(res, "millisecond")
+        res["hour"] = hour(default)
+        res["minute"] = minute(default)
+        res["second"] = second(default)
+        res["millisecond"] = millisecond(default)
+    else
+        if !haskey(res, "hour")
+            res["hour"] = 0
+        end
+        if !haskey(res, "minute")
+            res["minute"] = 0
+        end
+        if !haskey(res, "second")
+            res["second"] = 0
+        end
+        if !haskey(res, "millisecond")
+            res["millisecond"] = 0
+        end
+    end
```

Comment:
Is the code suppose to be inconsistent? Why are months and days used sometimes but not others?
```
julia> zdt = ZonedDateTime(DateTime(2014,2,3), TimeZone("Europe/Warsaw"))
2014-02-03T00:00:00+01:00

julia> parsedate("", default=zdt)
2014-02-03T00:00:00+01:00

julia> parsedate("2013", default=zdt)
2013-01-01T00:00:00+01:00
```